src/tlp_lib/smartcontracts/MockSC.py

```diff
@@ -1,58 +1,67 @@
 from datetime import datetime
 from typing import Any, Self
 
-from tlp_lib.protocols import GCTLP_Encrypted_Message, TLP_Digest, TLP_Digests
-from tlp_lib.smartcontracts.protocols import SC_Coins, SC_ExtraTime, SC_Solution, SC_Solutions, SC_UpperBounds
+from tlp_lib.protocols import GCTLP_Encrypted_Message, GCTLP_Encrypted_Messages, GCTLPInterface, TLP_Digest, TLP_Digests
+from tlp_lib.smartcontracts.protocols import SC_Coins, SC_UpperBounds
 
 
 class MockSC:
     commitments: TLP_Digests
     start_time: int
     upper_bounds: SC_UpperBounds
     coins: SC_Coins
-    solutions: SC_Solutions = []
-    initial_timestamp: int
+    solutions: GCTLP_Encrypted_Messages = []
+    gctlp: GCTLPInterface
     helper_id: Any
-    extra_time: SC_ExtraTime
 
     def initiate(
         self,
         coins: SC_Coins,
-        start_time: int,
-        extra_time: SC_ExtraTime,
         upper_bounds: SC_UpperBounds,
+        gctlp: GCTLPInterface,
         helper_id: Any,
     ) -> Self:
         self.coins = coins
-        self.start_time = start_time
-        self.extra_time = extra_time
+        self.start_time = int(datetime.now().timestamp())
         self.upper_bounds = upper_bounds
+        self.gctlp = gctlp
         self.helper_id = helper_id
         self.commitments = []
         self.solutions = []
-        self.initial_timestamp = int(datetime.now().timestamp())
         return self
 
     def add_solution(self, solution: GCTLP_Encrypted_Message, witness: TLP_Digest):
         time = int(datetime.now().timestamp())
-        self.solutions.append((solution, witness, time))
+        assert time < self.start_time + self.upper_bounds[-1]
```

Comment:
> such that they can be registered in SC by time t_j

the definition of Register_DTLP does say this but in a "it's in the interest of the TPH to do this" way, maybe it can be subtly rephrased.

The definition of Verify_DTLP is where the explicit checking happens.